app/grandchallenge/algorithms/views.py

```diff
@@ -555,6 +556,19 @@ def algorithm(self):
 
     def get_queryset(self):
         queryset = super().get_queryset()
+
+        def get_annotation(key):
```

Comment:
Is there a reason for defining a function here rather than doing it inline?

Reply:
For some reason, doing it inline did not work. I always got a TypeError:

![image](https://user-images.githubusercontent.com/30069334/140356408-9f3d8ba1-77f0-4e10-a538-4d2781a6f0a8.png)


`TypeError: QuerySet.annotate() received non-expression(s): {'probability-covid-19': <django.db.models.expressions.Subquery object at 0x7fc70ce41e80>}.`

Reply:
You;re not unpacking the dictionary there, how about `.annotate(interface.slug=Subquery(...))`?

Reply:
I think that you're missing a `*` before the `{` to unpack the dictionary. I'm also thinking that you could do this in one annotate call by building up the dictionary (eg with `{"slug1": ..., "slug2": ...}`) and then calling `queryset.annotate` only once after the for loop. However, thinking about it I'm wondering how performant it is as it will be doing 1 subquery per interface, we can always try it and see.

Reply:
Ah right, unpacking does the trick ;-) Annotating in one call also work, but it does not make a performance difference (same number of db calls). It's a bit easier to read though, so II pushed that change. In terms of performance, it actually only does the subqueries when the outer queryset is evaluated, so it does not add any extra db calls, as far as I can tell.